app/models/article.rb

```diff
@@ -35,7 +35,9 @@ def name
   end
 
   def path
-    if published?
+    if published? && short_path
```

Comment:
I think `short_path` can be as simple as just an attr on Article.

Then Article can have an `after_save` that creates a Redirect using `short_path` and `path`.

Make sense?